cy_ioctl.go

```diff
@@ -789,6 +803,7 @@ func (d *Device) sdpcmProcessRxPacket(buf []byte) (payloadOffset, plen uint32, h
 		plen = (uint32(hdr.Size) - payloadOffset) | uint32(itf)<<31
 
 	case whd.ASYNCEVENT_HEADER:
+		d.debug("sdpcmProcessRxPacket:ASYNC_HEADER")
 		const totalHeaderSize = whd.SDPCM_HEADER_LEN + whd.BDC_HEADER_LEN
```

Comment:
Getting Slice access panics here during initialization; 
some async event is yielding too large of a payload?

Reply:
I don't follow the comment/question.  Might be too late at night :)

Is it the added d.debug() that's exposing the panic?

Reply:
You may find that the firmware works most of the time, but once in a while it'll panic in this switch branch with a slice out of bounds access. I added the debug to catch the exact witch branch where it happens to better understand where it's happening.

Reply:
I suspect there's a bug somewhere here. Not only is there a OOB but the resulting async event is not triggering the LINK_UP, which is why we never succesfully connect to the AP. 
I've looked over the pointer arithmetic in the ASYNC_HEADER branch but I can't find a mistake if there is one.